engine/src/main/java/com/datatorrent/stram/util/AbstractWritableAdapter.java

```diff
@@ -56,6 +63,9 @@ public void readFields(DataInput arg0) throws IOException
       ois.close();
     }
     catch (Exception e) {
+      final Path path = Files.createTempFile("dt-", ".ser");
+      logger.error("Failed to de-serialize {}. Writing raw data to {}.", this.getClass().getName(), path, e);
```

Comment:
This may cause security concerns. Consider using an additional flag when logging the data.